src/components/PeopleFilters.tsx

```diff
@@ -1,96 +1,122 @@
+import { useSearchParams } from 'react-router-dom';
+import { SearchLink } from './SearchLink';
+
 export const PeopleFilters = () => {
-  return (
-    <nav className="panel">
-      <p className="panel-heading">Filters</p>
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sexParam = searchParams.get('sex') || 'all';
+  const currentCenturies = searchParams.getAll('centuries');
+  const queryParam = searchParams.get('query') || '';
 
-      <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">
-          All
-        </a>
-        <a className="" href="#/people?sex=m">
-          Male
-        </a>
-        <a className="" href="#/people?sex=f">
-          Female
-        </a>
-      </p>
+  const getActive = (currentValue: string, expectedValue: string) => {
+    return currentValue === expectedValue ? 'is-active' : '';
+  };
 
-      <div className="panel-block">
-        <p className="control has-icons-left">
-          <input
-            data-cy="NameFilter"
-            type="search"
-            className="input"
-            placeholder="Search"
-          />
+  const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = event.target.value;
 
-          <span className="icon is-left">
-            <i className="fas fa-search" aria-hidden="true" />
-          </span>
-        </p>
-      </div>
+    if (inputValue.length > 0) {
+      setSearchParams({
+        ...Object.fromEntries(searchParams),
+        query: inputValue,
+      });
+    } else {
+      const newParams = { ...Object.fromEntries(searchParams) };
 
-      <div className="panel-block">
-        <div className="level is-flex-grow-1 is-mobile" data-cy="CenturyFilter">
-          <div className="level-left">
-            <a
-              data-cy="century"
-              className="button mr-1"
-              href="#/people?centuries=16"
-            >
-              16
-            </a>
+      delete newParams.query;
 
-            <a
-              data-cy="century"
-              className="button mr-1 is-info"
-              href="#/people?centuries=17"
-            >
-              17
-            </a>
+      setSearchParams(newParams);
+    }
+  };
 
-            <a
-              data-cy="century"
-              className="button mr-1 is-info"
-              href="#/people?centuries=18"
-            >
-              18
-            </a>
+  const toggleCentury = (century: string) => {
+    const updatedCenturies = currentCenturies.includes(century)
+      ? currentCenturies.filter(centuries => centuries !== century)
+      : [...currentCenturies, century];
 
-            <a
-              data-cy="century"
-              className="button mr-1 is-info"
-              href="#/people?centuries=19"
-            >
-              19
-            </a>
+    return updatedCenturies.length > 0 ? updatedCenturies : null;
```

Comment:
The `toggleCentury` function returns `null` if `updatedCenturies` is empty. This might cause issues when setting search parameters, as `null` might not be the expected value. Consider returning an empty array or handling `null` appropriately when setting search parameters.